### PR TITLE
Feat/ProjectComponent database model is implemented

### PR DESCRIPTION
**Pull request by tvolk131**

### CODE REVIEW

**Review by brettkirk — DISMISSED** [review]

**Review by spoonlanding — APPROVED** [review]

**Review by brettkirk — APPROVED** [review]

File: db/models/projectComponent.js [hunk 1]
```diff
@@ -1,30 +1,152 @@
...
+ProjectComponent.getByProject = (projectId) => {
+  return Project.getById(projectId)
+    .then(() => {
+      return ProjectComponent.model.findAll({
+        where: {projectId}
+      });
+    });
+};
+
+// TODO - Implement this method
```
Comment (olegtrofimov1) [inline db/models/projectComponent.js:133]: Please do not leave old commented out code. Basically your codebase is the final version of your book that you want to send out to the world. You don't want little notes in the margins and highlights all over the place.  This commented out code makes it more difficult to read through this code base and creates appearance of this code still being in development, not yet ready for prod.

File: server/app.js [hunk 2]
```diff
@@ -10,18 +10,26 @@ app.use(middleware.morgan('dev'));
...
-app.set('view engine', 'ejs');
 
 app.use(middleware.auth.session);
 app.use(middleware.passport.initialize());
 app.use(middleware.passport.session());
-app.use(middleware.flash());
 
-app.use(express.static(path.join(__dirname, '../public')));
+// Serve static files
+app.get('*/bundle.js', (req, res) => {
```
Comment (olegtrofimov1) [inline server/app.js:19]: This seems weird. Why do you need to do this? Shouldn't this be handled by express.static?

File: server/middleware/passport.js [hunk 3]
```diff
@@ -28,173 +29,158 @@ passport.deserializeUser((id, done) => {
...
   }));
 
-passport.use('google', new GoogleStrategy({
-  clientID: config.Google.clientID,
-  clientSecret: config.Google.clientSecret,
-  callbackURL: config.Google.callbackURL
-},
-  (accessToken, refreshToken, profile, done) => getOrCreateOAuthProfile('google', profile, done))
-);
+// passport.use('google', new GoogleStrategy({
```
Comment (olegtrofimov1) [inline server/middleware/passport.js:96]: Please delete all this commented out code.

File: server/middleware/passport.js [hunk 4]
```diff
@@ -28,173 +29,158 @@ passport.deserializeUser((id, done) => {
...
-        done(error, null);
-      })
-      .catch(() => {
-        done(null, false, req.flash('signupMessage', 'An account with this email address already exists.'));
       });
   }));
 
 passport.use('local-login', new LocalStrategy({
-  usernameField: 'email',
+  usernameField: 'username',
```
Comment (olegtrofimov1) [inline server/middleware/passport.js:71]: Changes like this should be in a different PR. Please do not try to fix bugs or refactor in the same PR you are adding features.

File: server/routes/auth.js [hunk 5]
```diff
@@ -3,50 +3,27 @@ const middleware = require('../middleware');
 
 const router = express.Router();
 
-router.route('/')
-  .get(middleware.auth.verify, (req, res) => {
-    res.render('index.ejs');
-  });
+router.get('/', middleware.auth.verify);
```
Comment (olegtrofimov1) [inline server/routes/auth.js:6]: This looks like part of a refactor, in the future please do this in its own PR.

File: test/database/projectComponent.js [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+      .then((newUser) => {
+        userTwo = newUser;
+        return Project.create(project);
+      })
+      .then((newProject) => {
+        project = newProject;
+      });
+  });
+
+  describe('create', () => {
```
Comment (olegtrofimov1) [inline test/database/projectComponent.js:37]: This is really good

File: test/database/projectComponent.js [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+          expect(components[0].name).to.equal('test component');
+          return ProjectComponent.getByProject(project.id);
+        });
+    });
+    it('Should reject if userId does not map to an existing project', () => {
+      return expect(ProjectComponent.getByProject(1234)).to.be.rejectedWith('Project does not exist');
+    });
+  });
+
+  // TODO - Use these tests
```
Comment (olegtrofimov1) [inline test/database/projectComponent.js:190]: Wait why do you have this commented out? If this is not implemented yet you can simple create the skeleton. 

xdescribe('getByUser()', () => {
})

File: test/server/auth.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  beforeEach(() => {
+    return db.connection.clear();
+  });
+  it('Should be able to create new local users', (done) => {
+    let agent = chai.request.agent(app);
+    agent.post('/signup')
+      .send({username: 'test', password: 'test'})
+      .end((err, res) => {
+        expect(err).to.not.exist;
+        expect(res).to.have.cookie;
```
Comment (olegtrofimov1) [inline test/server/auth.js:17]: Can you also test that there isn't a cookie before signup?

**Review by olegtrofimov1 — COMMENTED** [review]
This was a pretty big PR. You can break this work into a few different PRs. Like 1) Fixing passport / adding bcrypt / 2) Adding static asset routes 3) Removing ejs templates. This would allow for a smaller PR that would be easier to review and would result in more bugs being caught. Ultimately you want it to be as easy as possible for your fellow contributors to review your code for mistakes. On a positive side, I do think you did a good job testing your project component.
